pkg/provisioner/ironic/ironic.go

```diff
@@ -362,13 +362,19 @@ func (p *ironicProvisioner) ValidateManagementAccess(credentialsChanged bool) (r
 	if ironicNode == nil {
 		p.log.Info("registering host in ironic")
 
+		deployInterface := "direct"
+		if p.host.Spec.LiveImage != nil {
```

Comment:
This should check for an empty string, too.

The precedence for the 2 image sources is a little confusing. In `NeedsProvisioning()` the `Image` is checked first so the `LiveImage` is only used if there is no regular image. Here, the `LiveImage` may override the `Image`. Similarly in `imageURL` on line 1076, the live image settings take precedence. However, if both `Image` and `LiveImage` are set, the `switch` statement on line 423 will apply the `Image` (because that rule comes first in the switch statement), and it would incorrectly use the "ramdisk" deploy interface set here.

It might help if we defined a function or two to give us the deploy settings for a host, so we only have to manage the precedence in one place.

Reply:
Thanks, that's a good idea, I was thinking they should be mutually exclusive so the order didn't matter, but until we actually implement such validation it will - I'll rework to use a function so it's consistent.